compiler/rustc_lint/src/lints.rs

```diff
@@ -960,10 +961,11 @@ impl AddToDiagnostic for NonBindingLetSub {
             rustc_errors::SubdiagnosticMessage,
         ) -> rustc_errors::SubdiagnosticMessage,
     {
+        let prefix = if self.default_binding_mode { "" } else { "let " };
```

Comment:
the comment on `default_binding_mode` says
>  At present, this is false only for destructuring assignment.

I don't like how that doesn't strongly imply that this truly is for destructuring assignment. This makes the code a bit harder to understand and potentially breaks in the future. I'm not sure what the nicest way to address this would be. If you agree.

This is also wrong and broken for non-trivial patterns like `(_, _) = (Droppy, Droppy)`, but oh well, those don't even lint today, so Not Your Problem. I'm gonna try fixing these missing lints myself, so I think it's fine to merge this as-is.

Reply:
update the code to use `assign_desugar`, maybe a better choice?

Reply:
This pr seems forget to update comments for `AssignDesugar`
https://github.com/chenyukang/rust/commit/8cf35643106bba09b5d6c71ceac74dc58573f371